detox/src/configuration/composeDeviceConfig.js

```diff
@@ -29,10 +36,24 @@ function validateUtilBinaryPaths({ errorBuilder, rawDeviceConfig }) {
  */
 function composeDeviceConfig({ errorBuilder, rawDeviceConfig, cliConfig }) {
   validateType({ errorBuilder, rawDeviceConfig });
+  validateAppLaunchArgs({ errorBuilder, rawDeviceConfig });
 
   rawDeviceConfig.device = getValidatedDeviceName({ errorBuilder, rawDeviceConfig, cliConfig });
   delete rawDeviceConfig.name;
 
+  if (cliConfig.appLaunchArgs) {
+    rawDeviceConfig.launchArgs = _.chain({})
+      .thru(() => parse(cliConfig.appLaunchArgs, {
+        configuration: {
+          'short-option-groups': false,
+        },
+      }))
+      .omit(['_', '--'])
+      .defaults(rawDeviceConfig.launchArgs)
+      .omitBy(value => value === false)
```

Comment:
Will this not also excessively omit values intentionally set to `false`?
```js
launchArgs: {
   verbose: false,
   someCustomDebug: false,
}
```

etc.

Reply:
@d4vidi, there is no reason to set those values to false since anyway they become converted to strings. If you want "false", you write `"false"`, not `false`.

The key idea about this `false` was to enable dropping launchArgs from CLI **(!)**, e.g. `--no-someCustomDebug` will just throw out the corresponding key-value from the default `launchArgs` map.

What we can do, however, is to throw an error on an attempt to use non-strings in that object (except for `false`, obviously).

Reply:
I assume we can close that — also I've added an extra validation to validate against non-string values in that launchArgs object, so this situation you have described now is just impossible.

Reply:
Yes, I spotted a `--no-someCustomDebug` equivalent in your unit tests, and thus realized what you were trying to do. Anyways, seems like we're good to go!